block-producer/src/state_view/mod.rs

```diff
@@ -33,28 +40,42 @@ impl<S> DefaultStateView<S>
 where
     S: Store,
 {
-    pub fn new(store: Arc<S>) -> Self {
+    pub fn new(
+        store: Arc<S>,
+        verify_tx_proofs: bool,
+    ) -> Self {
```

Comment:
This is probably fine for now - but I wonder if we should pass a config struct into this (either new or existing).

Reply:
I could add a struct as follows: 

```rust
pub struct DefaultStateViewConfig<S> {
    store: Arc<S>,
    verify_tx_proofs: bool,
}

pub struct DefaultStateView<S> {
    store: Arc<S>,

    /// Enables or disables the verification of transaction proofs in `verify_tx`
    verify_tx_proofs: bool,

    /// The account ID of accounts being modified by transactions currently in the block production
    /// pipeline. We currently ensure that only 1 tx/block modifies any given account (issue: #186).
    accounts_in_flight: Arc<RwLock<BTreeSet<AccountId>>>,

    /// The nullifiers of notes consumed by transactions currently in the block production pipeline.
    nullifiers_in_flight: Arc<RwLock<BTreeSet<Digest>>>,
}

impl<S> DefaultStateView<S>
where
    S: Store,
{
    pub fn new(config: DefaultStateViewConfig<S>) -> Self {
        Self {
            store: config.store,
            verify_tx_proofs: config.verify_tx_proofs,
            accounts_in_flight: Arc::new(RwLock::new(BTreeSet::new())),
            nullifiers_in_flight: Arc::new(RwLock::new(BTreeSet::new())),
        }
    }
}
```

But I don't fully understand why we would want to do so, I might not have understood your request.

Reply:
> This is probably fine for now - but I wonder if we should pass a config struct into this (either new or existing).

To drop my 2 cents. IMO we should give all the settings as configuration variables, wrapped in a [`watch`](https://docs.rs/tokio/latest/tokio/sync/watch/index.html). This would make it possible to have config hot reload.

Reply:
> I could add a struct as follows

I don't think we need to do this yet - but if we need to pass in more config settings into the state view, a dedicated struct may be needed.

> To drop my 2 cents. IMO we should give all the settings as configuration variables, wrapped in a [`watch`](https://docs.rs/tokio/latest/tokio/sync/watch/index.html). This would make it possible to have config hot reload.

Let's create an issue to enable hot config reloads. I don't think this is a priority right now - but would be a cool feature to support down the line.